images/router/haproxy/conf/haproxy-config.template

```diff
@@ -214,7 +214,7 @@ backend be_edge_http_{{$cfgIdx}}
   {{ end }}
   http-request set-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
                 {{ range $idx, $endpoint := endpointsForAlias $cfg $serviceUnit }}
-  server {{$endpoint.ID}} {{$endpoint.IP}}:{{$endpoint.Port}} check inter 5000ms cookie {{$endpoint.ID}}
+  server {{$endpoint.ID}} {{$endpoint.IP}}:{{$endpoint.Port}} check inter 5000ms cookie {{$endpoint.IDhash}}
```

Comment:
Ram,
Thanks, I will make those changes.
phil
On 04/04/2016 02:29 PM, Ram Ranganathan wrote:
>
> In images/router/haproxy/conf/haproxy-config.template 
> <https://github.com/openshift/origin/pull/8334#discussion_r58423905>:
>
> > @@ -214,7 +214,7 @@ backend be_edge_http_{{$cfgIdx}}
> >    {{ end }}
> >    http-request set-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
> >                  {{ range $idx, $endpoint := endpointsForAlias $cfg $serviceUnit }}
> > -  server {{$endpoint.ID}} {{$endpoint.IP}}:{{$endpoint.Port}} check inter 5000ms cookie {{$endpoint.ID}}
> > +  server {{$endpoint.ID}} {{$endpoint.IP}}:{{$endpoint.Port}} check inter 5000ms cookie {{$endpoint.IDhash}}
>
> Needs to be |server {{$endpoint.IDHash}} {{$endpoint.IP}} ...|
> The server |id| is set to |{{$endpoint.ID}}| - it should match the 
> IDHash - that value from the cookie is used to pick a specific server.
>
> Also |IDHash| seems more consistent as a name than |IDhash|.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/openshift/origin/pull/8334/files/d5a28c792427c49fed6676cc043d4a13b621d62c#r58423905>
>